x-pack/plugins/observability_solution/apm/server/deprecations/apm_user_role.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  SecurityGetRoleMappingResponse,
+  SecurityGetUserResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+import type {
+  CoreSetup,
+  DeprecationsDetails,
+  DocLinksServiceSetup,
+  ElasticsearchClient,
+  GetDeprecationsContext,
+} from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+import type { DeprecationApmDeps } from '.';
+import { deprecations } from '../lib/deprecations';
+
+const APM_USER_ROLE_NAME = 'apm_user';
+const getKibanaPrivilegesDocumentationUrl = (branch: string) => {
+  return `https://www.elastic.co/guide/en/kibana/${branch}/kibana-privileges.html`;
+};
+
+export async function getDeprecationsInfo(
+  { esClient }: GetDeprecationsContext,
+  core: CoreSetup,
+  apmDeps: DeprecationApmDeps
+) {
+  const client = esClient.asCurrentUser;
+  const { docLinks } = core;
+  const { security } = apmDeps;
+
+  // Nothing to do if security is disabled
+  if (!security?.license.isEnabled()) {
+    return [];
+  }
+
+  return [
+    ...(await getUsersDeprecations(client, apmDeps, docLinks)),
+    ...(await getRoleMappingsDeprecations(client, apmDeps, docLinks)),
+  ];
+}
+
+async function getUsersDeprecations(
+  client: ElasticsearchClient,
+  apmDeps: DeprecationApmDeps,
+  docLinks: DocLinksServiceSetup
+): Promise<DeprecationsDetails[]> {
+  const strings = {
+    title: i18n.translate('xpack.apm.deprecations.apmUser.title', {
+      defaultMessage: `The "{apmUserRoleName}" role has been removed: check user roles`,
+      values: { apmUserRoleName: APM_USER_ROLE_NAME },
+    }),
+    message: i18n.translate('xpack.apm.deprecations.apmUser.description', {
+      defaultMessage: `The "{apmUserRoleName}" has been removed and this cluster has users with the removed role.`,
+      values: { apmUserRoleName: APM_USER_ROLE_NAME },
+    }),
+    manualSteps: (usersRoles: string) => [
+      i18n.translate('xpack.apm.deprecations.apmUser.manualStepOne', {
+        defaultMessage: `Go to Management > Security > Users to find users with the "{apmUserRoleName}" role.`,
+        values: { apmUserRoleName: APM_USER_ROLE_NAME },
+      }),
+      i18n.translate('xpack.apm.deprecations.apmUser.manualStepTwo', {
+        defaultMessage:
+          `Remove the "{apmUserRoleName}" role from all users and add the built-in "viewer" role.` +
+          ` The affected users are: {usersRoles}.`,
+        values: { apmUserRoleName: APM_USER_ROLE_NAME, usersRoles },
+      }),
+    ],
+  };
```

Comment:
Maybe something like this? I'm not sure if it's appropriate to use `deprecated` in place of `removed`.

> ### Check for users assigned the deprecated "apm_user" role
> 
> The "apm_user" role has been deprecated. Remove the "apm_user" role from affected users in this cluster including: userA, userB, userC .
> 
> **How to fix**
> 
> 1. Go to Management > Security > Users to find users with the "apm_user" role.
> 2. Remove the "apm_user" role from all users and add the built-in "viewer" role.

Reply:
These changes LGTM! 🦭 👍